corehq/apps/hqadmin/reports.py

```diff
@@ -1166,3 +1168,70 @@ def _create_row(self, log, *args, **kwargs):
         row = super(DeviceLogSoftAssertReport, self)._create_row(log, *args, **kwargs)
         row.append(log.domain)
         return row
+
+
+class CommCareVersionReport(AdminReport, ElasticTabularReport):
+    slug = "commcare_version"
+    name = ugettext_lazy("CommCare Version")
+
+    @property
+    def headers(self):
+        versions = get_all_versions()
+        headers = DataTablesHeader(
+            DataTablesColumn(_("Project"))
+        )
+        for version in versions:
+            headers.add_column(DataTablesColumn(version))
+        return headers
+
+    @property
+    def rows(self):
+        versions = get_all_versions()
+
+        def get_query(version):
+            return {
+                "query": {
+                    "bool": {
+                        "must": [
+                            {
+                                "range": {
+                                    "received_on": {
+                                        "gte": "now-90d",
+                                        "lte": "now"
+                                    }
+                                }
+                            },
+                            {
+                                "query_string": {
+                                    "default_field": "appVersion",
+                                    "query": "*" + version + "*"
+                                }
+                            }
+                        ]
+                    }
+                },
+                "aggs": {
+                    "by_domain": {
+                        "terms": {
+                            "field": "domain"
+                        }
+                    }
+                }
+            }
+
+        rows = {}
+        for domain in es_domain_query(show_stats=False)['hits']['hits']:
+            domain_name = domain['_source']['name']
+            rows.update({domain_name: [domain_name] + [0] * len(versions)})
+
+        for idx, version in enumerate(versions, 2):
```

Comment:
@emord 
I wanted use nested aggregation but to returns correct data I must use ```script```. I talked with Cory and Simon and they want disable this possibility from ES on HQ.

My idea was: add ```script``` which take from ```appVersion``` property only version (by regexp) and then group by this version, so in data I will get something like:

```
{
    "aggregations": {
        "by_domain": {
            "buckets": [
                {
                    "key": "domain_name",
                    "doc_count": "some number",
                    "by_version": {
                        "buckets": [
                            {
                                "key": "2.22.1",
                                "doc_count": "some number"
                            },
                            {
                                "key": "2.22.2",
                                "doc_count": "some number"
                            },
                            ...
                        ]
                    }
                },
                ...
            ]
        }
    }
}
```

Do you have any suggestion how I can do this without ```script```?